app/views/add-reduction.html

```diff
@@ -0,0 +1,156 @@
+        </legend>
+        <div class="form-date" id="reduction_end_day">
+          <div class="form-group form-group-day">
+            <label class="form-label" for="end-day">Day</label>
+            <input class="form-control" id="end-day" name="red_end_day" type="number" pattern="[+-]?[0-9]{1,2}" min="1" max="31">
+          </div>
+          <div class="form-group form-group-month" id="reduction_end_month">
+            <label class="form-label" for="end-month">Month</label>
+            <input class="form-control" id="end-month" name="red_end_month" type="number" pattern="[+-]?[0-9]{1,2}" min="1" max="12">
+          </div>
+          <div class="form-group form-group-year" id="reduction_end_year">
+            <label class="form-label" for="end-year">Year</label>
+            <input class="form-control" id="end-year" name="red_end_year" type="number" pattern="[+-]?[0-9]{1,4}" min="0" max="">
+          </div>
+        </div>
+      </fieldset>
+    </div>
+
+    <div class="form-group">
+      <label class="form-label-bold" for="textarea">Additional notes</label>
+      <textarea class="form-control form-control-3-4" name="notes" id="textarea" rows="5"></textarea>
+    </div>
+    <input type="hidden" name="reasonForReductionId" id="hiddenId" value="0"> 
+    {% include "includes/csrf-hidden-input.html"%} 
+    <input class="button" type="submit" value="Add reduction">
+  </form>
+
+</main>
+{% endblock %} {% block body_end %}
+<script type="text/javascript">
+  var refData = {{ referenceData | dump | safe }}
+
+  var chosenReduction
+
+  var populateHours = function () {    
+    var value = ''
+    var hiddenIdValue
+
+    var chosenReductionIndex = document.getElementById("select-box").selectedIndex
+
+    chosenReduction = refData[chosenReductionIndex]
+    
+    if (chosenReduction.allowancePercentage !== null) {
+      document.getElementById('hours').value = chosenReduction.allowanceHours
+      document.getElementById('hours').setAttribute('readonly', true)
+    } else {
+      document.getElementById('hours').value = value
+      document.getElementById('hours').removeAttribute('readonly')
+    }
+    
+    if (chosenReduction.maxAllowancePercentage !== null) {
+      document.getElementById('hours').setAttribute('max', chosenReduction.maxAllowanceHours)
+    }
+
+    document.getElementById('hiddenId').value = hiddenIdValue
+  }
+
+  var calculateExpiry = function () {    
+    var chosenReductionIndex = document.getElementById("select-box").selectedIndex
+    chosenReduction = refData[chosenReductionIndex]
+
+    var startMonth = document.getElementById("start-month").value
+    var readonly = false
+    var endDay = ''
+    var endMonth = ''
+    var endYear = ''
+
+    if (chosenReduction.monthsToExpiry !== null) {
+      if (Number(startMonth) > 0 && Number(startMonth) < 13) {
+        readonly = true
+        endDay = document.getElementById("start-day").value
```

Comment:
Validate that the day exists in the new month? e.g 6 months from 31st of March is 31st September which does not exist

Reply:
Yea, so that is going to cause issues. We might be able to make use of JS Date function for this and then split into string.

Reply:
Done.